backend/endpoints/rooms.py

```diff
@@ -435,3 +435,76 @@ async def get_members(org_id: str, room_id: str):
             message="Room members retrieved successfully",
         ),
     )
+
+
+@router.patch(
+    "/org/{org_id}/members/{member_id}/rooms/{room_id}",
+    response_model=ResponseModel,
+    status_code=status.HTTP_201_CREATED,
+    responses={
+        200: {"detail": {"room_id": "room_id"}},
+        424: {"detail": "ZC Core Failed"},
+    },
+ )
+async def update_room(
+    org_id: str, member_id: str, room_id: str, request: UpdateRoomRequest, background_tasks: BackgroundTasks
+ ):
+    """Updates a room between users.
+    Updates a room document to the database collection.
+    Returns the document id and updated info if the room is successfully updated
+    while publishing to the user sidebar in the background
+    Args:
+        org_id (str): A unique identifier of an organisation
+        request: A pydantic schema that defines the room request parameters
+        member_id: A unique identifier of the member updating the room
+    Returns:
+        HTTP_200_OK (room updated): {room_id}
+    Raises
+        HTTP_401_UNAUTHORIZED: not a member of room
+        HTTP_401_UNAUTHORIZED: must be an admin to update room
+        HTTP_424_FAILED_DEPENDENCY: room update unsuccessful
+    """
+    # Initialize Database
+    DB = DataStorage(org_id)
+
+    # Get room from database
+    room = await get_room(org_id=org_id, room_id=room_id)
+    data = request.dict()
+
+    # Get the members of the room, if None assign to empty dictionary
+    members = room.get("room_members") or {}
+    updater = members.get(str(member_id))
+
+    if updater == None:
+        raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="not a member of room",
+    )
+
+    if updater.get("role") != Role.ADMIN:
+        raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="must be an admin to update room.",
+    )
+
+    # Insert members into data request to avoid deletion of all members
+    data['members'] = members
+
+    # updates the room data in the db collection
+    update_response = await DB.update(
+        settings.ROOM_COLLECTION, document_id=room_id, data=data
+    )  
+
+    if update_response and update_response.get("status_code", None) is None:
+        room_id = {"room_id": update_response.get("data").get("object_id")}
+        
```

Comment:
why do you need to change the room id. shouldn't it be the same as the one originally fetched?

Reply:
That's a honest mistake. I'm suppose to remove that line.